packages/wrangler/src/sites.tsx

```diff
@@ -175,27 +196,31 @@ export async function syncAssets(
       logger.log(`Skipping - already uploaded.`);
     }
 
-    // remove the key from the set so we know what we've already uploaded
+    // Remove the key from the set so we know what we've already uploaded
     namespaceKeys.delete(assetKey);
 
-    // prevent causing different manifest keys on windows
-    const maifestKey = urlSafe(
+    // Prevent different manifest keys on windows
+    const manifestKey = urlSafe(
       path.relative(siteAssets.assetDirectory, absAssetFile)
     );
-    manifest[maifestKey] = assetKey;
+    manifest[manifestKey] = assetKey;
   }
 
+  // Add the last (potentially only) bucket to the batch
+  uploadBuckets.push(uploadBucket);
+
   // keys now contains all the files we're deleting
   for (const key of namespaceKeys) {
     logger.log(`Deleting ${key} from the asset store...`);
   }
 
-  await Promise.all([
-    // upload all the new assets
-    putKVBulkKeyValue(accountId, namespace, toUpload),
-    // delete all the unused assets
-    deleteKVBulkKeyValue(accountId, namespace, Array.from(namespaceKeys)),
-  ]);
+  // sequentially upload each bucket
```

Comment:
FWIW @threepointone @petebacondarwin: I compared uploading a 150MiB site via this sequential solution, and a Promise.all approach:
```
  const bucketsToPut = [];
  for (const bucket of uploadBuckets) {
    bucketsToPut.push(putKVBulkKeyValue(accountId, namespace, bucket));
  }
  await Promise.all(bucketsToPut);
```
The sequential solution took `51 sec` while the parallel solution took `38 sec`, could be worth looking into if folks complain it's too slow

Reply:
I can't recollect why I made this sequential, probably to avoid saturating the network. We could make this parallel, and consider adding limited concurrency later if people complain with extremely large payloads (like, GBs)